contracts/IdentityRegistryBridger.sol

```diff
@@ -310,4 +358,99 @@ contract IdentityRegistryBridger is ERC721TokenReceiver {
         }
         _locked = 1;
     }
+
+    /// @dev Links service Ids with registered 8004 agent Ids.
+    /// @param numServices Number of services to link.
+    /// @return agentIds Set of 8004 agent Ids.
+    function linkServiceIdAgentIds(uint256 numServices) external returns (uint256[] memory agentIds) {
```

Comment:
we said that we are also going to have a method enabling us to have an update by providing all relevant info fetched off-cain to avoid this gas intensive method